### PR TITLE
Gradle: support direct file dependencies

### LINKED ISSUE #3066: How to add local jar file with gradle?
standard way:
dependencies {
   compile files("path/to/jar")
}
not work
if starting with quakusDev, we see java.lang.NoClassDeffoundError: oracle/jdbc/driver/OracleDriver

### PR DESCRIPTION
**Pull request by aloubyansky**
Fixes #3066

### CODE REVIEW

File: devtools/gradle/src/main/java/io/quarkus/gradle/builder/QuarkusModelBuilder.java [hunk 1]
```diff
@@ -234,7 +235,14 @@ private void collectFirstMetDeploymentDeps(Set<ResolvedDependency> dependencies,
 
     private void collectDependencies(ResolvedConfiguration configuration,
             LaunchMode mode, Project project, Map<ArtifactCoords, Dependency> appDependencies) {
-        for (ResolvedArtifact a : configuration.getResolvedArtifacts()) {
+        final Set<ResolvedArtifact> artifacts = configuration.getResolvedArtifacts();
+        Set<File> artifactFiles = null;
+        // if the number of artifacts is less then the number of files then probably
```
Comment (gsmet) [inline devtools/gradle/src/main/java/io/quarkus/gradle/builder/QuarkusModelBuilder.java:240]:
```suggestion
        // if the number of artifacts is less than the number of files then probably
```

**Review by gsmet — DISMISSED** [review]
Spotted a small typo, please dismiss when fixed.

**Review by glefloch — APPROVED** [review]
LGTM

**Comment by aloubyansky** [comment]
@gsmet it's waiting on you now. Thanks.

**Review by gastaldi — APPROVED** [review]
LGTM

**Comment by gastaldi** [comment]
@aloubyansky can you rebase this PR just to make CI happy?

**Pull request merged**